packages/grpc-js/src/subchannel.ts

```diff
@@ -505,6 +509,14 @@ export class Subchannel {
         }
         this.session = null;
         this.stopKeepalivePings();
+        /* If the backoff timer has already ended by the time we get to the
+         * TRANSIENT_FAILURE state, we want to immediately transition out of
+         * TRANSIENT_FAILURE as though the backoff timer is ending right now */
+        if (!this.backoffTimeout.isRunning()) {
+          process.nextTick(() => {
+            this.handleBackoffTimer();
```

Comment:
Hm. I think I would've liked it slightly better if there was a way to restart the `backoffTimeout` with a deadline of "now", which would've essentially streamlined this.

Reply:
Fundamentally, what's happening here is that the backoff timer already triggered, but the result of the timer was deferred until this state transition. It's important here that the `backoffTimeout` is not reset, and that it keeps any backoff it has accumulated.